tests/lexer/tstrnumlits.nim

```diff
@@ -102,50 +44,84 @@ assertAST dedent """
       Ident "\'wrap"""":
   -38383839292839283928392839283928392839283.928493849385935898243e-50000'wrap
 
-doAssert $1234.56'wrap == "[[1234.56]]":
-  "unable to properly account for context with suffixed numeric literals"
-
-# verify that the i64, f32, etc builtin suffixes still parse correctly
-
-const expectedF32: float32 = 123.125
-
-doAssert 123.125f32 == expectedF32:
-  "failing to support non-quoted legacy f32 floating point suffix"
-
-doAssert 123.125'f32 == expectedF32
-
-doAssert 123.125e0'f32 == expectedF32
-
-# if not a built-in suffix, test that we can parse as a user supplied suffix
-
-proc `'f9`(number: string): S =   # proc starts with 'f' just like 'f32'
-  result = ("[[" & number & "]]").S
-
-proc `'d9`(number: string): S =   # proc starts with 'd' just like the d suffix
-  result = ("[[" & number & "]]").S
-
-proc `'i9`(number: string): S =   # proc starts with 'i' just like 'i64'
-  result = ("[[" & number & "]]").S
-
-proc `'u9`(number: string): S =   # proc starts with 'u' just like 'u8'
-  result = ("[[" & number & "]]").S
-
-doAssert $1234.56'wrap == $1234.56'f9
-doAssert $1234.56'wrap == $1234.56'd9
-doAssert $1234.56'wrap == $1234.56'i9
-doAssert $1234.56'wrap == $1234.56'u9
-
-doAssert lispReprStr(1234.56'u9) == """(DotExpr (StrLit "1234.56") (Ident "\'u9"))""":
-  "failed to properly build AST for suffix that starts with u"
-
-doAssert 1E2 == 100:
-  "lexer not handling upper-case exponent"
-
-doAssert 1.0E2 == 100.0:
-  "lexer not handling upper-case exponent with float"
-
-doAssert 1e2 == 100:
-  "lexer not handling lower-case exponent"
-
-doAssert $1E2'wrap == "[[1E2]]":
-  "lexer not passing exponent (e) marker with original case"
+template main =
+  block: # basic suffix usage
+    proc `'wrap`(number: string): string =
+      result = "[[" & number & "]]"
+    template `'twrap`(number: string): untyped =
+      number.`'wrap`
+    proc extraContext(): string =
+      22.40'wrap
+    proc `*`(left, right: string): string =
+      result = left & "times" & right
+    proc `+`(left, right: string): string =
+      result = left & "plus" & right
+
+    doAssert 1'wrap == "[[1]]"
+    doAssert -1'wrap == "[[-1]]":
+      "unable to resolve a negative integer-suffix pattern"
+    doAssert lispReprStr(-1'wrap) == """(DotExpr (StrLit "-1") (Ident "\'wrap"))"""
+    doAssert 12345.67890'wrap == "[[12345.67890]]"
+    doAssert 1'wrap*1'wrap == "[[1]]times[[1]]":
+      "unable to resolve an operator between two suffixed numeric literals"
+    doAssert 1'wrap+ -1'wrap == "[[1]]plus[[-1]]":
```

Comment:
```suggestion
    doAssert 1'wrap + -1'wrap == "[[1]]plus[[-1]]":
```

(doesn't nim warn about this?)

Reply:
Yes it does. I went ahead and added a version that is consistent to be complete. Is there a way to test for a warning?

Reply:
there will be once i make a PR for https://github.com/nim-lang/Nim/issues/8803 and convince araq that this is indeed a good feature (as simple as `let ret = compilesWithContext(expr)`), until then testing for warnings can be done but is not as flexible, via a testament discard section with nimout etc); in your case, it's ok to not check for the warning (it's mostly irrelevant for this test file) and use both with correct spacing and without for completeness.